### PR TITLE
file download example

### PR DESCRIPTION
**Pull request by nddipiazza**
add an example of how to download a file from grpc.

### CODE REVIEW

File: examples/src/main/java/io/grpc/examples/filedownload/FileDownloadServer.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          int length;
+          while ((length = bis.read(buffer, 0, bufferSize)) != -1) {
+            responseObserver.onNext(
+              DataChunk.newBuilder().setData(ByteString.copyFrom(buffer, 0, length)).build()
+            );
+          }
+          responseObserver.onCompleted();
+        }
+      } catch (IOException e) {
+        e.printStackTrace();
```
Comment (carl-mastrangelo) [inline examples/src/main/java/io/grpc/examples/filedownload/FileDownloadServer.java]: This should probably be an RPC Error.    Somethign like `responseObserver.onError` ?

File: examples/src/main/proto/file_download.proto [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2015 The gRPC Authors
```
Comment (carl-mastrangelo) [inline examples/src/main/proto/file_download.proto]: 2018

**Comment by dapengzhang0** [comment]
@nddipiazza There is [a good place](https://github.com/saturnism/grpc-java-by-example) for various such examples mentioned in [grpc-ecosystem](https://github.com/grpc-ecosystem/awesome-grpc/blob/master/readme.md#res-examples). And they are accepting pull requests.

**Comment by dapengzhang0** [comment]
@nddipiazza Thank you for your pull request. However, we would recommend adding it to https://github.com/saturnism/grpc-java-by-example rather than here.

**Pull request closed without merging**